### PR TITLE
[GATE] Personate system token

### PR DESCRIPTION
**Pull request by emiteze**
## Issue Description

Moove keeps authorId on almost every domain and uses Authorization header to search for a user on our db. Since now an request can be valid without the Authorization but having X-Charles-Token, all moove endpoints were returning error since they we're not founding users based on x-charles-token.

## Solution

Create a user every time a system token is created (personate) and change moove endpoints that search for users on db based on Authorization to support X-Charles-Token as well.

### CODE REVIEW

**Review by monicaribeiro — APPROVED** [review]

File: gate/internal/domain/system_token.go [hunk 1]
```diff
@@ -36,3 +36,15 @@ type SystemToken struct {
 	LastUsedAt    *time.Time
 	Author        string
 }
+
+func (systemToken SystemToken) CreateUserFromSystemToken() User {
+	return User{
+		ID:            uuid.New(),
+		Name:          systemToken.Name,
+		PhotoUrl:      "",
```
Comment (barbararochazup) [inline gate/internal/domain/system_token.go:44]: It's necessary this field?

File: gate/internal/domain/user.go [hunk 2]
```diff
@@ -29,5 +29,6 @@ type User struct {
 	PhotoUrl  string
 	Email     string
 	IsRoot    bool
+	SystemTokenId string
```
Comment (barbararochazup) [inline gate/internal/domain/user.go:32]: SystemTokenId should be  uuid.UUID instead of string?

File: gate/internal/repository/user_repository.go [hunk 3]
```diff
@@ -70,6 +72,18 @@ func (userRepository userRepository) GetByEmail(email string) (domain.User, erro
...
 }
 
+func (userRepository userRepository) Create(user domain.User) (domain.User, error) {
+	user.ID = uuid.New()
+	userToSave := mapper.UserDomainToModel(user)
+
+	res := userRepository.db.Model(models.User{}).Create(&userToSave)
+
+	if res.Error != nil {
+		return domain.User{}, handleSystemTokenError("Create user failed", "UserRepository.Create.create", res.Error, logging.InternalError)
```
Comment (barbararochazup) [inline gate/internal/repository/user_repository.go:82]: Should be a handleUserError?

**Review by barbararochazup — CHANGES_REQUESTED** [review]
Maybe to do some tests in practical cases using system token. Because unit tests have only test using authorization.

**Pull request merged**